### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by GrikTad**

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -83,7 +113,7 @@ def eval_fn(self, current_epoch):
 
             total_loss += loss.item()
             tk.set_postfix({"Loss": "%6f" % float(total_loss / (t + 1))})
-
+        print(f"f1:{self.f1_metric(labels,logits)}, recall:{self.rec_metric(labels,logits)},precision:{self.prec_metric(labels,logits)}")
```
Comment (Barsegh-A) [inline src/utils.py:116]: this will calculate only for the last batch of validation set, we need to collect all results and calculate metrics on the whole validation set
